src/main/java/minefantasy/mfr/block/BlockBloomery.java

```diff
@@ -112,25 +115,27 @@ public boolean onBlockActivated(World world, BlockPos pos, IBlockState state, En
 
 			ItemStack held = player.getHeldItem(hand);
 
-			// Hammer
+			/// Hammer
 			if (!player.isSwingInProgress && tile.tryHammer(player)) {
 				return true;
 			}
-			// Light
-			int l = ItemLighter.tryUse(held, player);
-			if (!tile.isActive() && !tile.hasBloom()) {
-				if (!held.isEmpty() && l != 0) {
-					player.playSound(SoundEvents.ITEM_FLINTANDSTEEL_USE, 1.0F, 1.0F);
-					if (world.isRemote)
-						return true;
-					if (l == 1 && tile.light(player)) {
-						held.damageItem(1, player);
+			/// Ignition
+			if (held.getItem() instanceof ItemFlintAndSteel || held.getItem() instanceof ILighter) {
+				if (!tile.getIsActive() && tile.getResult() != ItemStack.EMPTY) {
+					int uses = ItemLighter.tryUse(held, player);
+					if (uses != 0) // 1 for ignition, -1 for a failed attempt, 0 for a null input or for an item that needs to bypass normal ignition
+					{
+						player.playSound(SoundEvents.ITEM_FLINTANDSTEEL_USE, 1.0F, 1.0F);
+						if (uses == 1 && !world.isRemote) {
+							held.damageItem(1, player);
+							fireItUp(world, pos, state);
+						}
 					}
 					return true;
 				}
 			}
-			// GUI
-			if (!world.isRemote && !tile.isActive() && !tile.hasBloom()) {
+			/// GUI
+			if (!world.isRemote && !tile.getIsActive() && !tile.hasBloom()) {
```

Comment:
I was originally planning on adding getters and setters for all the variables in each tile entity, but I scrapped that idea once I got a better idea of how everything was working and realized that getters and setters for everything was completely overkill. I changed the name of a couple of functions to be in line with this before deciding to scrap it - I can change it back if you would prefer it the old way

Reply:
I think it would be good to change them back 

I just think isLit() is slightly easier to understand, because its essentially a question in the method, answered by the returned boolean